resources/js/features/settings/components/LocaleSectionCard/LocaleSectionCard.tsx

```diff
@@ -0,0 +1,78 @@
+import { useLaravelReactI18n } from 'laravel-react-i18n';
+import type { FC } from 'react';
+
+import {
+  BaseFormControl,
+  BaseFormDescription,
+  BaseFormField,
+  BaseFormItem,
+  BaseFormLabel,
+} from '@/common/components/+vendor/BaseForm';
+import {
+  BaseSelect,
+  BaseSelectContent,
+  BaseSelectItem,
+  BaseSelectTrigger,
+  BaseSelectValue,
+} from '@/common/components/+vendor/BaseSelect';
+import { usePageProps } from '@/common/hooks/usePageProps';
+
+import { SectionFormCard } from '../SectionFormCard';
+import { useLocaleSectionForm } from './useLocaleSectionForm';
+
+export const LocaleSectionCard: FC = () => {
+  const { auth } = usePageProps<App.Community.Data.UserSettingsPageProps>();
+
+  const { t } = useLaravelReactI18n();
+
+  const { form, mutation, onSubmit } = useLocaleSectionForm({
+    locale: auth?.user.locale?.length ? auth.user.locale : 'en_US',
+  });
+
+  return (
+    <SectionFormCard
+      t_headingLabel={t('Locale')}
+      formMethods={form}
+      onSubmit={onSubmit}
+      isSubmitting={mutation.isPending}
+    >
+      <div className="flex flex-col gap-7 @container @xl:gap-5">
+        <BaseFormField
+          control={form.control}
+          name="locale"
+          render={({ field }) => (
+            <BaseFormItem className="flex w-full flex-col gap-1 @xl:flex-row">
+              <BaseFormLabel htmlFor="locale-select" className="text-menu-link @xl:mt-4 @xl:w-2/5">
+                {t('Current Locale')}
+              </BaseFormLabel>
+
+              <div className="flex flex-grow flex-col gap-1">
+                <BaseFormControl>
+                  <BaseSelect onValueChange={field.onChange} defaultValue={field.value}>
+                    <BaseSelectTrigger id="locale-select">
+                      <BaseSelectValue />
+                    </BaseSelectTrigger>
+
+                    <BaseSelectContent>
+                      {/* These labels should not be localized. */}
+                      <BaseSelectItem value="en_US">{'English (US)'}</BaseSelectItem>
+                      <BaseSelectItem value="pt_BR">{'Português (Brasil)'}</BaseSelectItem>
+                    </BaseSelectContent>
+                  </BaseSelect>
+                </BaseFormControl>
+
+                <BaseFormDescription>
+                  {t(
+                    "Most of the website is still untranslated. If you'd like to help with translations, you can learn more about how to contribute",
+                  )}{' '}
+                  <a href="#">{t('here')}</a>
+                  {t('.')}
```

Comment:
String construction is generally not translatable. Would it make sense to have the link be placeholder or shortcode?
`how to contribute :here.`
`how to contribute [url=X]here[/url]`

Reply:
Thankfully, this is a solved problem. We just need a complex translation component.

I'm currently working on this in a separate branch and am already done with implementation. The only thing left is to write comprehensive tests.

My implementation is based on that of another popular i18n library ([ref](https://react.i18next.com/latest/trans-component#using-with-react-components)).

Here's how it works:
```tsx
<p>
  <Trans i18nKey="profile" values={{ name: 'John' }}>
    Welcome to <strong>{name}</strong>'s profile
  </Trans>
</p>
```
```json
"profile": "Welcome to <0>:name</0>'s profile",
```

This outputs the following HTML:
```html
<p>Welcome to <strong>John</strong>'s profile</p>
```

Now in another locale, this can be written as:
```json
"profile": "<0>:name</0> is the profile you're viewing"
```
And the user of that locale would see:
```html
<p><strong>John</strong> is the profile you're viewing</p>
```

Reply:
Addressed in https://github.com/RetroAchievements/RAWeb/pull/2774.